src/main/java/org/opensearch/knn/index/codec/jvector/JVectorRandomAccessReader.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.jvector;
+
+import io.github.jbellis.jvector.disk.RandomAccessReader;
+import io.github.jbellis.jvector.disk.ReaderSupplier;
+import lombok.extern.log4j.Log4j2;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.IOContext;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.util.IOUtils;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Log4j2
+public class JVectorRandomAccessReader implements RandomAccessReader {
+    private final byte[] buffer = new byte[Long.BYTES];
+    private final IndexInput indexInputDelegate;
+    private volatile boolean closed = false;
+
+    public JVectorRandomAccessReader(IndexInput indexInputDelegate) {
+        this.indexInputDelegate = indexInputDelegate;
+    }
+
+    @Override
+    public void seek(long offset) throws IOException {
+        indexInputDelegate.seek(offset);
+    }
+
+    @Override
+    public long getPosition() throws IOException {
+        return indexInputDelegate.getFilePointer();
+    }
+
+    @Override
+    public int readInt() throws IOException {
+        indexInputDelegate.readBytes(buffer, 0, Integer.BYTES);
+        // Replace ByteArray.getInt with ByteBuffer implementation
+        return ByteBuffer.wrap(buffer, 0, Integer.BYTES).order(ByteOrder.BIG_ENDIAN).getInt();
+    }
+
+    @Override
+    public float readFloat() throws IOException {
+        return Float.intBitsToFloat(readInt());
+    }
+
+    @Override
+    public long readLong() throws IOException {
+        indexInputDelegate.readBytes(buffer, 0, Long.BYTES);
+        return ByteBuffer.wrap(buffer, 0, Long.BYTES).order(ByteOrder.BIG_ENDIAN).getLong();
+    }
+
+    @Override
+    public void readFully(byte[] bytes) throws IOException {
+        indexInputDelegate.readBytes(bytes, 0, bytes.length);
+    }
+
+    @Override
+    public void readFully(ByteBuffer buffer) throws IOException {
+        indexInputDelegate.readBytes(buffer.array(), buffer.arrayOffset(), buffer.remaining());
+    }
+
+    @Override
+    public void readFully(long[] vector) throws IOException {
+        for (int i = 0; i < vector.length; i++) {
```

Comment:
these may be faster if you get the bytes into a BB then extract the longs.  Same with all the others.

Reply:
all good